pkg/engine/internal/worker/thread.go

```diff
@@ -59,20 +70,20 @@ NextTask:
 		// ensures that the context of tasks written to respCh are bound to the
 		// lifetime of the thread, but can also be canceled by the scheduler.
 		req := readyRequest{
-			Context:  ctx,
+			Context:  context.Background(),
```

Comment:
`readyRequest.Context` is always `context.Background` now, so I think we can remove the field from the struct now